src/main/java/sirius/biz/jobs/batch/file/VirtualFileExtractionJob.java

```diff
@@ -0,0 +1,116 @@
+                                                                .withMessage(NLS.fmtr(
+                                                                        "VirtualFileExtractionJob.skippingOverwrite")
+                                                                                .set("targetPath", targetFile.path())
+                                                                                .format()));
+                                          return false;
+                                      }
+
+                                      try {
+                                          if (targetFile.exists() && shouldOverwriteExisting) {
+                                              process.log(ProcessLog.info()
+                                                                    .withMessage(NLS.fmtr(
+                                                                            "VirtualFileExtractionJob.extractingFile")
+                                                                                    .set("filePath", filePath)
+                                                                                    .set("targetPath",
+                                                                                         targetFile.path())
+                                                                                    .set("fileSize",
+                                                                                         NLS.formatSize(data.size()))
+                                                                                    .format()));
+                                          } else {
+                                              process.log(ProcessLog.info()
+                                                                    .withMessage(NLS.fmtr(
+                                                                            "VirtualFileExtractionJob.overwritingFile")
+                                                                                    .set("filePath", filePath)
+                                                                                    .set("targetPath",
+                                                                                         targetFile.path())
+                                                                                    .set("fileSize",
+                                                                                         NLS.formatSize(data.size()))
+                                                                                    .format()));
+                                          }
+
+                                          targetFile.consumeStream(data.openStream(), data.size());
+                                      } catch (IOException e) {
+                                          process.handle(e);
+                                      }
+
+                                      process.log(ProcessLog.info()
```

Comment:
i'd output the progrss just as title and also maybe ratelimit this (TaskContext should have a shouldUpdateTitle or the like)